File: ecommerce_backend/models/category_model.js
import { Schema, model } from 'mongoose';
//import { v1 } from 'uuid';
//import { genSaltSync, hashSync } from 'bcrypt';

const categorySchema = Schema({
    title: { type: String, required: [true, "title is required, pls provide"] },
    description: { type: String,  requried: [true, "description is required, pls provide"], },
    updatedOn: { type: Date, },
    createdOn: { type: Date, }

});

categorySchema.pre('save', function (next) {
    this.updatedOn = new Date();
    this.createdOn = new Date();

    next(); // if next has string that mean it has error
});

categorySchema.pre(['update', 'findOneAndUpdate'], function (next) {
    const update = this.getUpdate();
    delete update._id;//?

    this.updatedOn = new Date();
    next();
});

const categoryModel = model('Category', categorySchema);

//module.exports = categoryModel;
export default categoryModel;

File: ecommerce_frontend/lib/data/models/category_model.dart
class CategoryModel {
  String? sId;
  late String title;
  late String description;
  String? updatedOn;
  String? createdOn;

  CategoryModel({
    this.sId,
    required this.title,
    required this.description,
    this.updatedOn,
    this.createdOn,
  });

  CategoryModel.fromJson(Map<String, dynamic> json) {
    sId = json['_id'];
    title = json['title'];
    description = json['description'];
    updatedOn = json['updatedOn'];
    createdOn = json['createdOn'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = <String, dynamic>{};
    data['_id'] = sId;
    data['title'] = title;
    data['description'] = description;
    data['updatedOn'] = updatedOn;
    data['createdOn'] = createdOn;
    return data;
  }
}
